fuel/app/classes/oauth.php

```diff
@@ -10,24 +9,24 @@ public static function validate_post()
 			$signature  = \Input::post('oauth_signature', '');
 			$timestamp  = (int) \Input::post('oauth_timestamp', 0);
 			$nonce      = \Input::post('oauth_nonce', false);
-			$lti_config = LtiLaunch::config();
+			$lti_config = LtiLaunch::config_from_request();
 
 			if (empty($signature)) throw new \Exception('Authorization signature is missing.');
 			if (empty($nonce)) throw new \Exception('Authorization fingerprint is missing.');
-			if (\Input::post('oauth_consumer_key') !== $lti_config['key']) throw new \Exception('Authorization signature failure.');
+			if (\Input::post('oauth_consumer_key') !== $lti_config['key']) throw new \Exception('Authorization key failure.');
 			if ($timestamp < (time() - $lti_config['timeout'])) throw new \Exception('Authorization signature is too old.');
 
 			$hasher   = new \Eher\OAuth\HmacSha1(); // THIS CODE ASSUMES HMACSHA1, could be more versetile, but hey
 			$consumer = new \Eher\OAuth\Consumer(null, $lti_config['secret']);
-			$request  = \Eher\OAuth\Request::from_consumer_and_token($consumer, null, 'POST', \Uri::current(), \Input::post());
+			$request  = \Eher\OAuth\Request::from_consumer_and_token($consumer, null, 'POST', \Uri::main(), \Input::post());
```

Comment:
main() is more accurate for the purpose of signing - FuelPHP can internally re-route to another uri - signing requires validating off the original url the server received.